File: backend/src/main/java/vn/edu/ut/pbms/constant/BookingStatus.java
package vn.edu.ut.pbms.constant;

/**
 * Status enum for Booking entity.
 * PENDING = chờ xác nhận, CONFIRMED = đã xác nhận,
 * CHECKED_IN = đã check-in, CANCELLED = đã hủy, EXPIRED = đã hết hạn.
 */
public enum BookingStatus {
    PENDING,
    CONFIRMED,
    CHECKED_IN,
    CANCELLED,
    EXPIRED
}

File: frontend/src/constants/booking-status.ts
// ---------------------------------------------
// Booking Status Constants
// Trạng thái đơn đặt chỗ trước
// ---------------------------------------------

export enum BookingStatus {
  PENDING = "PENDING",
  CONFIRMED = "CONFIRMED",
  CHECKED_IN = "CHECKED_IN",
  CANCELLED = "CANCELLED",
  EXPIRED = "EXPIRED",
}

export const BOOKING_STATUS_LABELS: Record<BookingStatus, string> = {
  [BookingStatus.PENDING]: "Chờ xác nhận",
  [BookingStatus.CONFIRMED]: "Đã xác nhận",
  [BookingStatus.CHECKED_IN]: "Đã check-in",
  [BookingStatus.CANCELLED]: "Đã hủy",
  [BookingStatus.EXPIRED]: "Hết hạn",
};
